src/shared/utils/radarUtils.ts

```diff
@@ -0,0 +1,143 @@
+import * as d3 from 'd3';
+import * as R from 'ramda';
+import { color } from '../../theme';
+
+// custom random number generator, to make random sequence reproducible
+// source: https://stackoverflow.com/questions/521295
+let seed = 42;
+export const random = () => {
+  const x = Math.sin(seed++) * 10000;
+  return x - Math.floor(x);
+};
+
+export const random_between = (min: number, max: number) => {
+  return min + random() * (max - min);
+};
+
+export const normal_between = (min: number, max: number) => {
+  return min + (random() + random()) * 0.5 * (max - min);
+};
+
+export const polar = (cartesian: { x: number; y: number }) => {
+  const x = cartesian.x;
+  const y = cartesian.y;
+  return {
+    t: Math.atan2(y, x),
+    r: Math.sqrt(x * x + y * y),
+  };
+};
+
+export const cartesian = (polar: { r: number; t: number }) => {
+  return {
+    x: polar.r * Math.cos(polar.t),
+    y: polar.r * Math.sin(polar.t),
+  };
+};
+
+export const bounded_interval = (value: number, min: number, max: number) => {
+  const low = Math.min(min, max);
+  const high = Math.max(min, max);
+  return Math.min(Math.max(value, low), high);
+};
+
+export const bounded_ring = (polar: { t: number; r: number }, r_min: number, r_max: number) => {
+  return {
+    t: polar.t,
+    r: bounded_interval(polar.r, r_min, r_max),
+  };
+};
+
+export const bounded_box = (
+  point: { x: number; y: number },
+  min: { x: number; y: number },
+  max: { x: number; y: number }
+) => {
+  return {
+    x: bounded_interval(point.x, min.x, max.x),
+    y: bounded_interval(point.y, min.y, max.y),
+  };
+};
+
+export const translate = (x: number, y: number) => {
+  return 'translate(' + x + ',' + y + ')';
+};
+
+export const getRadarScale = (): number => {
+  const availableRadarWidth = window.innerWidth - 411;
+  const basicRadarWidth = 1450;
```

Comment:
basic radar width and height can be defined somewhere as constants `*.constants.ts`
other values which are constants can be moved there as well

Reply:
Done